tests/test_space_ops.py

```diff
@@ -2,14 +2,87 @@
 
 import numpy as np
 
-from manim import cartesian_to_spherical, spherical_to_cartesian
+from manim.utils.space_ops import *
+from manim.utils.space_ops import shoelace, shoelace_direction
+
+
+def test_rotate_vector():
+    vec = np.array([0, 1, 0])
+    rotated = rotate_vector(vec, np.pi / 2)
+    assert np.round(rotated[0], 5) == -1.0
+    assert not np.round(rotated[1:], 5).any()
+
+
+def test_rotation_matrices():
+    ang = np.pi / 6
+    ax = np.array([1, 1, 1])
+    assert np.all(
+        np.round(rotation_matrix(ang, ax, True), 5)
+        == np.round(
+            np.array(
+                [
+                    [0.91068, -0.24402, 0.33333, 0.0],
+                    [0.33333, 0.91068, -0.24402, 0.0],
+                    [-0.24402, 0.33333, 0.91068, 0.0],
+                    [0.0, 0.0, 0.0, 1.0],
+                ]
+            ),
+            5,
+        )
+    )
+    assert np.all(
+        np.round(rotation_about_z(np.pi / 3), 5)
+        == np.array(
+            [
+                [0.5, -0.86603, 0.0],
+                [0.86603, 0.5, 0.0],
+                [0.0, 0.0, 1.0],
+            ]
+        )
+    )
+    assert np.all(
+        np.round(z_to_vector(np.array([1, 2, 3])), 5)
+        == np.array(
+            [
+                [0.96362, 0.0, 0.26726],
+                [-0.14825, 0.83205, 0.53452],
+                [-0.22237, -0.5547, 0.80178],
+            ]
+        )
+    )
+
+
+def test_angle_of_vector():
+    assert angle_of_vector(np.array([1, 1, 1])) == np.pi / 4
+    assert (
+        np.round(angle_between_vectors(np.array([1, 1, 1]), np.array([-1, 1, 1])), 5)
+        == 1.23096
+    )
+
+
+def test_center_of_mass():
+    assert np.all(center_of_mass([[0, 0, 0], [1, 2, 3]]) == np.array([0.5, 1.0, 1.5]))
+
+
+def test_line_intersection():
```

Comment:
Can you test that
- a ValueError is properly raised when the lines don't line in the xy plane;
- the case where the lines are parallel
- the case where the lines aren't parallel, but the intersection occurs outside the specified line segments

Reply:
> the case where the lines aren't parallel, but the intersection occurs outside the specified line segments

This is not implemented, and neither was the previous implementation. Tbh, idt it should be.

Reply:
Hmm, what if the lines are equal too?

Reply:
Its pretty much the same as parallel lines, do you think we need to test for it too?

Reply:
I think more tests are (almost) never a bad thing